raiden/routing.py

```diff
@@ -129,3 +198,105 @@ def get_best_routes(
         route_state = RouteState(partner_address, channel_state_id)
         available_routes.append(route_state)
     return available_routes
+
+
+def get_best_routes_pfs(
+        chain_state: ChainState,
+        token_network_id: typing.TokenNetworkID,
+        from_address: typing.InitiatorAddress,
+        to_address: typing.TargetAddress,
+        amount: int,
+        previous_address: typing.Optional[typing.Address],
+        config: Dict[str, Any],
+) -> Tuple[bool, List[RouteState]]:
+    pfs_path = '{}/api/v1/{}/paths'.format(
+        config['pathfinding_service_address'],
+        to_checksum_address(token_network_id),
+    )
+    payload = {
+        'from': to_checksum_address(from_address),
+        'to': to_checksum_address(to_address),
+        'value': amount,
+        'max_paths': config['pathfinding_max_paths'],
+    }
+
+    # check that the response is successful
+    try:
+        response = requests.get(pfs_path, params=payload, timeout=DEFAULT_HTTP_REQUEST_TIMEOUT)
+    except requests.RequestException as re:
+        log.info(
+            'Could not connect to Pathfinding Service',
+            request=pfs_path,
+            parameters=payload,
+            exception_=re,
+        )
+        return False, []
+
+    # check that the response contains valid json
+    try:
+        response_json = response.json()
+    except ValueError as ve:
+        log.info(
+            'Pathfinding Service returned invalid JSON',
+            exception_=ve,
+        )
+        return False, []
+
+    if response.status_code != 200:
+        log_info = {
+            'error_code': response.status_code,
+        }
+
+        error = response_json.get('errors')
+        if error is not None:
+            log_info['pfs_error'] = error
+
+        log.info(
+            'Pathfinding Service returned error code',
+            **log_info,
+        )
+        return False, []
+
+    if response_json.get('result') is None:
+        log.info(
+            'Pathfinding Service returned unexpected result',
+            result=response_json,
+        )
+        return False, []
+
+    paths = []
+    network_statuses = views.get_networkstatuses(chain_state)
+    for path_object in response_json['result']:
+        path = path_object['path']
```

Comment:
More nitpicks: Both of there could also fail if the response doesn't have the expected structure.

Reply:
I guess we should add some schema for these results in order to be able to check them properly. But I'd prefer to do that in another PR.